crates/hir_def/src/type_ref.rs

```diff
@@ -188,11 +188,22 @@ impl TypeRef {
                         is_varargs = param.dotdotdot_token().is_some();
                     }
 
-                    pl.params().map(|p| p.ty()).map(|it| TypeRef::from_ast_opt(ctx, it)).collect()
+                    pl.params()
+                        .map(|p| (p.pat(), p.ty()))
+                        .map(|it| {
+                            let type_ref = TypeRef::from_ast_opt(ctx, it.1);
+                            let name = if it.0.is_some() {
+                                Some(it.0.unwrap().syntax().text().to_string())
+                            } else {
+                                None
+                            };
```

Comment:
This pattern of "if X has a value, transform it -- otherwise None" is something I have encountered before but always come up with this awkward check. What's the idiomatic way to do this?

Reply:
`it.0.map(|it| it.syntax().text().to_string())`

Reply:
Ah thank you! Code doesn't need it since an earlier refactor but I'll keep that in mind